src/index.ts

```diff
@@ -0,0 +1,775 @@
+/**
+ * A Result model, loosely based on the Rust result type.
+ *
+ * @module
+ */
+
+/**
+ * `Result<T, E>` is used for returning and propagating errors.
```

Comment:
(personal gripe) For an industry that discourages single letter variable names we sure seem to love single letter placeholders for generics…